numba/targets/mathimpl.py

```diff
@@ -355,11 +355,24 @@ def hypot_u64_impl(context, builder, sig, args):
 @lower(math.hypot, types.Float, types.Float)
 def hypot_float_impl(context, builder, sig, args):
     def hypot(x, y):
+        # This particular algorithm avoids intermediate overflow in the
+        # computation.  See
+        # http://www.johndcook.com/blog/2010/06/02/whats-so-hard-about-finding-a-hypotenuse/
+        x, y = abs(x), abs(y)
+        if x < y:
+            x, y = y, x
         if math.isinf(x):
-            return abs(x)
-        elif math.isinf(y):
-            return abs(y)
-        return math.sqrt(x * x + y * y)
+            return x
+        if math.isinf(y):
```

Comment:
No, because `inf` has the priority over `nan` here (i.e. `hypot(inf, nan) == hypot(nan, inf) == inf`).
(also, `math.isfinite` only exists on 3.x, but we could probably work that around)